include/deal.II/matrix_free/shape_info.templates.h

```diff
@@ -433,39 +433,38 @@ namespace internal
                 for (unsigned int f = 0; f < n_faces; ++f)
                   {
                     const unsigned int n_face_orientations =
-                      dim == 2 ?
-                        2 :
-                        (2 *
-                         reference_cell.face_reference_cell(f).n_vertices());
+                      reference_cell.n_face_orientations(f);
 
                     const unsigned int n_q_points_face =
                       quad_face[quad_face.size() == 1 ? 0 : f].size();
 
-                    for (unsigned int o = 0; o < n_face_orientations; ++o)
+                    for (types::geometric_orientation orientation = 0;
+                         orientation < n_face_orientations;
+                         ++orientation)
                       {
-                        const auto offset =
+                        // like elsewhere, the default in the matrix-free code
+                        // is 0, not 1, so flip the first bit
+                        const unsigned char mf_orientation = orientation ^ 1;
```

Comment:
Thanks for leaving the comment. Is that something you eventually want to unify as well?

Reply:
Yes - once we are done unifying the orientation scheme the matrix-free encoding will be the default and we can get rid of this line (and use `orientation` in the next couple of lines).